src/compile/mark/area.ts

```diff
@@ -11,80 +14,107 @@ export namespace area {
   export function properties(model: UnitModel) {
     // TODO Use Vega's marks properties interface
     let p: any = {};
+    const config = model.config();
 
-    const orient = model.config().mark.orient;
-    if (orient !== undefined) {
-      p.orient = { value: orient };
-    }
+    const _orient = orient(config.mark.orient);
+    if (_orient) { p.orient = _orient; }
+    
+    const _x = x(model.encoding().x, model.scaleName(X), model.stack());
+    if (_x) { p.x = _x; }
+
+    const _x2 = x2(model.encoding().x, model.scaleName(X), model.stack(), config.mark.orient);
+    if (_x2) { p.x2 = _x2; }
+    
+    const _y = y(model.encoding().y, model.scaleName(Y), model.stack());
+    if (_y) { p.y = _y; }
+ 
+    const _y2 = y2(model.encoding().y, model.scaleName(Y), model.stack(), config.mark.orient);
+    if (_y2) { p.y2 = _y2; }
 
-    const stack = model.stack();
-    const xFieldDef = model.encoding().x;
+    applyColorAndOpacity(p, model);
+    applyMarkConfig(p, model, ['interpolate', 'tension']);
+    return p;
+  }
+  
+  function orient(orient: string): VgValueRef {
+    if (orient) {
+      return { value: orient };
+    }
+    return undefined;
+  }
+  
+  function x(fieldDef: FieldDef, scaleName: string, stack: StackProperties): VgValueRef {
```

Comment:
I think we need to pass in scale and if (!scale), then don't output `scaleName`.